frappe/email/inbox.py

```diff
@@ -54,8 +54,10 @@ def mark_as_seen_unseen(name, action):
 
 	for name in json.loads(names or []):
 		uid, seen_status, email_account = frappe.db.get_value(
-			"Communication", name, ["ifnull(uid, -1)", "ifnull(seen, 0)", "email_account"]
+			"Communication", name, ["uid", "seen", "email_account"]
 		)
+		uid = uid if uid else -1
```

Comment:
L61 skips the iteration if uid is not found or is `-1` - so we can remove this as well :)